dpnp/dpnp_iface_manipulation.py

```diff
@@ -1994,6 +2127,138 @@ def hstack(tup, *, dtype=None, casting="same_kind"):
+        Axis along which to insert `values`. If `axis` is ``None`` then `arr`
+        is flattened first.
+        Default: ``None``.
+
+    Returns
+    -------
+    out : dpnp.ndarray
+        A copy of `arr` with `values` inserted. Note that :obj:`dpnp.insert`
+        does not occur in-place: a new array is returned. If
+        `axis` is ``None``, `out` is a flattened array.
+
+    See Also
+    --------
+    :obj:`dpnp.append` : Append elements at the end of an array.
+    :obj:`dpnp.concatenate` : Join a sequence of arrays along an existing axis.
+    :obj:`dpnp.delete` : Delete elements from an array.
+
+    Notes
+    -----
+    Note that for higher dimensional inserts ``obj=0`` behaves very different
+    from ``obj=[0]`` just like ``arr[:, 0, :] = values`` is different from
+    ``arr[:, [0], :] = values``.
+
+    Examples
+    --------
+    >>> import dpnp as np
+    >>> a = np.array([[1, 1], [2, 2], [3, 3]])
+    >>> a
+    array([[1, 1],
+           [2, 2],
+           [3, 3]])
+    >>> np.insert(a, 1, 5)
+    array([1, 5, 1, 2, 2, 3, 3])
+    >>> np.insert(a, 1, 5, axis=1)
+    array([[1, 5, 1],
+           [2, 5, 2],
+           [3, 5, 3]])
+
+    Difference between sequence and scalars:
+
+    >>> np.insert(a, [1], [[1],[2],[3]], axis=1)
+    array([[1, 1, 1],
+           [2, 2, 2],
+           [3, 3, 3]])
+    >>> np.array_equal(np.insert(a, 1, [1, 2, 3], axis=1),
+    ...                np.insert(a, [1], [[1],[2],[3]], axis=1))
+    array(True)
+
+    >>> b = a.flatten()
+    >>> b
+    array([1, 1, 2, 2, 3, 3])
+    >>> np.insert(b, [2, 2], [5, 6])
+    array([1, 1, 5, 6, 2, 2, 3, 3])
+
+    >>> np.insert(b, slice(2, 4), [5, 6])
+    array([1, 1, 5, 2, 6, 2, 3, 3])
+
+    >>> np.insert(b, [2, 2], [7.13, False]) # dtype casting
+    array([1, 1, 7, 0, 2, 2, 3, 3])
+
+    >>> x = np.arange(8).reshape(2, 4)
+    >>> idx = (1, 3)
+    >>> np.insert(x, idx, 999, axis=1)
+    array([[  0, 999,   1,   2, 999,   3],
+           [  4, 999,   5,   6, 999,   7]])
+
+    """
+
+    dpnp.check_supported_arrays_type(arr)
+    params = _calc_parameters(arr, axis, obj, values)
+
+    if isinstance(obj, slice):
+        # turn it into a range object
+        indices = dpnp.arange(
+            *obj.indices(params.n),
+            dtype=dpnp.intp,
+            sycl_queue=params.exec_q,
+            usm_type=params.usm_type,
+        )
+    else:
+        # need to copy obj, because indices will be changed in-place
+        indices = dpnp.array(
+            obj, copy=True, sycl_queue=params.exec_q, usm_type=params.usm_type
+        )
+        if indices.dtype == dpnp.bool:
+            warnings.warn(
+                "In the future insert will treat boolean arrays and array-likes"
+                " as a boolean index instead of casting it to integers",
+                FutureWarning,
+                stacklevel=2,
+            )
+            indices = indices.astype(dpnp.intp)
+            # TODO: Code after warning period:
```

Comment:
It is currently aligned with the latest NumPy. If we update it to treat boolean array as boolean index, some test will fail when compared to current available NumPy versions. We can wait to make the change when NumPy does or implement the new approach and modify the tests to avoid direct comparison with NumPy.

Reply:
I see. I thought it is already done in the latest NumPy. Then I'm fine to keep it as you did, please ignore the comment.